gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/DataSource-pouchDB.js

```diff
@@ -0,0 +1,194 @@
+/*!
+Copyright 2016 OCAD university
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+/* eslint-env browser */
+/* eslint strict: ["error", "function"] */
+
+var fluid = fluid || require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+
+(function () {
+
+    "use strict";
+
+    // This pouchDB data source is to support the in-browser testing that uses pouchDB as the backend data storage.
+    // It overrides default setImpl() and getImpl() provided by "kettle.dataSource.URL" with gpii.pouch API calls.
+    // Note: This component requires an instance of gpii.pouch (see https://github.com/the-t-in-rtf/gpii-pouchdb/blob/GPII-1897/docs/pouchdb.md)
+    // to be instantiated beforehand.
+    fluid.defaults("gpii.dataSource.pouchDB", {
+        gradeNames: ["kettle.dataSource.URL"],
```

Comment:
Based on some experimentation I was doing to see if I could integrate `gpii.dataSource.pouchDB` as part of some non-GPII work, I believe this grade definition may need the following addition:
```
  readOnlyGrade: "gpii.dataSource.pouchDB"
```

Without it, the `readOnlyGrade` option value is `kettle.dataSource.URL` (set from the parent grade), which results in the writeable grade being composited from `gpii.dataSource.pouchDB` and  `kettle.dataSource.URL.writable` (rather than `gpii.dataSource.pouchDB.writable`) when `kettle.dataSource.getWriteable` is invoked to construct the dynamic writeable grade if the `writable` option is true. 

This means the parent `setImpl` invoker is used (`kettle.dataSource.URL.handle` instead of `gpii.dataSource.pouchDB.handle`) when doing `set` operations.

Reply:
Cheers @waharnum , @cindyli  - thanks for the reports on this situation, which were very helpful in thinking out a proper route for https://issues.fluidproject.org/browse/KETTLE-49 . In the end we ended up with a system based on the ContextAwareness API responding to a writable:true flag - this preserves the benefits of a resettable flag, as well as using an established API and producing a homogeneous system that looks the same to providers of writable grades regardless of where they are in the hierarchy.